### PR TITLE
[RFR] Fix clear filter

### LINKED ISSUE #650: Error on clean filter
How to reproduce the error:

- In a list resource, select a filter, and filter by a field (Ok)
- Remove the filter clicking in cross button (Ok)
- Go other resource (Ok)
- Go back to the previous resource where you apply the filter (BAD, the value is "undefined" for the filter).

The problem is in the REMOVE FILTER button. 

Query Output:

> GET /api/users?UserName=**undefined**&_sort=id&_order=DESC&_start=0&_end=10

### PR DESCRIPTION
**Pull request by djhi**
Fixes #650

### CODE REVIEW

**Comment by fzaninotto** [comment]
It'd be reaaaaally nice to have a test for this one. Besides, the code formatting is weird. What would prettier give on this one?

**Comment by djhi** [comment]
Not sure about prettier formatting for this yet. I will make a pr for prettier though. Agreed for tests, back to WIP

**Comment by djhi** [comment]
Should be good now !

File: e2e/tests/list.js [hunk 1]
```diff
@@ -4,53 +4,64 @@ import driver from '../chromeDriver';
...
         it('should display paginated list of available posts', async () => {
-            assert.equal(await ListPage.getNbPagesText(), '1-10 of 13');
+            assert.equal(await ListPagePosts.getNbPagesText(), '1-10 of 13');
         });
 
         it('should switch page when clicking on previous/next page buttons or page numbers', async () => {
-            await ListPage.nextPage();
-            assert.equal(await ListPage.getNbPagesText(), '11-13 of 13');
+            await ListPagePosts.nextPage();
+            await ListPagePosts.checkPagination('11-13 of 13');
```
Comment (fzaninotto) [inline e2e/tests/list.js]: There is no `assert` in `checkPagination`. It seems that you expect a timeout for this test to fail, which is not a good practice IMO. You should replace `checkPagination` by `getPagination` and do the assert here (as it was before).

File: package.json [hunk 2]
```diff
@@ -74,7 +74,7 @@
     "react-tap-event-plugin": "~2.0.1",
     "recompose": "~0.23.1",
     "redux": "~3.6.0",
-    "redux-form": "~6.6.3",
+    "redux-form": "~6.7.0",
```
Comment (fzaninotto) [inline package.json]: why? Seems unrelated.

File: src/reducer/resource/data.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ import {
  * }
  */
 
-const cacheDuration = 10 * 60 * 1000; // ten minutes
+const cacheDuration = 10 * 60 * 1; // ten minutes
```
Comment (fzaninotto) [inline src/reducer/resource/data.js]: typo?

File: src/reducer/resource/list/ids.js [hunk 4]
```diff
@@ -1,10 +1,62 @@
...
+const cacheDuration = 10 * 60 * 1; // ten minutes
+
+/**
+ * Add new records to the pool, and remove outdated ones.
+ *
+ * This is the equivalent of a stale-while-revalidate caching strategy:
+ * The cached data is displayed before fetching, and stale data is removed
+ * only once fresh data is fetched.
+ */
+const addRecords = (newRecords = [], oldRecords) => {
```
Comment (fzaninotto) [inline src/reducer/resource/list/ids.js]: This looks like a big copy-paste of `data.js`. Besides, why is it related to the clear filter bug?

File: src/reducer/resource/data.js (hunk 3 above)
Comment (djhi) [inline src/reducer/resource/data.js]: artefact for another bug hunt

File: src/reducer/resource/list/ids.js (hunk 4 above)
Comment (djhi) [inline src/reducer/resource/list/ids.js]: artefact for another bug hunt

File: package.json (hunk 2 above)
Comment (djhi) [inline package.json]: artefact from another bug hunt

File: e2e/tests/list.js [hunk 5]
```diff
@@ -3,54 +3,64 @@ import { By, until } from 'selenium-webdriver';
 import driver from '../chromeDriver';
 import listPageFactory from '../pages/ListPage';
 
-describe('List Page', () => {
-    const ListPage = listPageFactory('http://localhost:8083/#posts')(driver);
+describe.only('List Page', () => {
```
Comment (fzaninotto) [inline e2e/tests/list.js]: .only spotted 😱

File: e2e/pages/ListPage.js [hunk 6]
```diff
@@ -27,17 +27,24 @@ module.exports = (url) => (driver) => ({
...
+            .then(() => driver.sleep(250)); // let some time to redraw
+        });
     },
 
     getNbRows() {
         return driver.findElements(this.elements.recordRows)
             .then(rows => rows.length);
     },
 
+    checkPagination(expected) {
```
Comment (fzaninotto) [inline e2e/pages/ListPage.js]: not needed anymore?

**Pull request merged**